### PR TITLE
Remove deprecations, missing types, fix old code

### PR DESCRIPTION
**Pull request by kunicmarko20**
There should probably some changelog and upgrade note, right?

### CODE REVIEW

File: src/Test/EntityManagerMockFactory.php [hunk 1]
```diff
@@ -24,39 +24,39 @@
...
             ->disableOriginalConstructor()->getMock();
         $query->expects($test->any())->method('execute')->will($test->returnValue(true));
 
         if (Version::compare('2.5.0') < 1) {
             $entityManager = $test->getMockBuilder(EntityManagerInterface::class)->getMock();
-            $qb = $test->getMockBuilder(QueryBuilder::class)->setConstructorArgs([$entityManager])->getMock();
+            $queryBuilder = $test->getMockBuilder(QueryBuilder::class)->setConstructorArgs([$entityManager])->getMock();
         } else {
-            $qb = $test->getMockBuilder(QueryBuilder::class)->disableOriginalConstructor()->getMock();
+            $queryBuilder = $test->getMockBuilder(QueryBuilder::class)->disableOriginalConstructor()->getMock();
```
Comment (OskarStark) [inline src/Test/EntityManagerMockFactory.php]: You could use createMock here, couldn’t you?

**Comment by greg0ire** [comment]
Please fix lint issues

**Comment by greg0ire** [comment]
And conflicts :P

**Comment by kunicmarko20** [comment]
Not sure what should I write in changelog/upgrade note since https://github.com/sonata-project/SonataCoreBundle/pull/564 covered all that I would write.

**Pull request merged**